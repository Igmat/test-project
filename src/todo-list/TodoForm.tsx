import { ListItem, TextField } from '@material-ui/core';
import { observer } from 'mobx-react';
import React, { ChangeEventHandler, Component, FormEventHandler } from 'react';
import TodoState from './state';

export interface ITodoFormProps {

}
export interface ITodoFormState {
    name: string;
}
@observer
export default class TodoForm extends Component<ITodoFormProps, ITodoFormState> {
    state = {
        name: ''
    }
    private onChange: ChangeEventHandler<HTMLInputElement> = ev => {
        this.setState({ name: ev.target.value })
    }
    private onSubmit: FormEventHandler = ev => {
        ev.preventDefault();
        TodoState.add(this.state.name);
        this.setState({ name: '' })
    }
    render() {
        return (
            <ListItem role={undefined} dense >
                <form onSubmit={this.onSubmit}>
                    <TextField name="name" value={this.state.name} label="New TODO" onChange={this.onChange} />
                </form>
            </ListItem>
        )
    }
}
