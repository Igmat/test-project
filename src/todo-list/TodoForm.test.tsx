import React from 'react';
import renderer from 'react-test-renderer';
import TodoForm from './TodoForm';

test('TodoForm matches snapshot', () => {
  const tree = renderer
    .create(<TodoForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});