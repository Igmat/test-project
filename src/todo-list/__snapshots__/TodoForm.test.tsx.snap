// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm matches snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
  disabled={false}
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        New TODO
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
</li>
`;
